Middleware/DynamicJavaScriptMiddleware.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.IO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Hosting;
+using System.Threading.Tasks;
+
+public class DynamicJavaScriptMiddleware
+{
+    private readonly RequestDelegate _next;
+    private readonly IWebHostEnvironment _env;
+
+    public DynamicJavaScriptMiddleware(RequestDelegate next, IWebHostEnvironment env)
+    {
+        _next = next;
+        _env = env;
+    }
+
+    public async Task InvokeAsync(HttpContext context)
+    {
+        OSFLogo osfLogo = new OSFLogo();
+        ButtonSVG button = new ButtonSVG();
+        Webpage webpage = new Webpage();
+
+        context.Response.ContentType = "application/javascript";
+//        string jsContent = await ReadJavaScriptFileAsync("osf-logo.js");
+        string jsContent = await ReadJavaScriptFileAsync("webpage.js");
+        string svg = jsContent;
+
+        if (context.Request.Path.Value.EndsWith(".js"))
+        {
+	    switch (context.Request.Path.Value)
+	    {
+		case "/foo.js": svg = await osfLogo.SVG(svg, _env);
+		    await context.Response.WriteAsync(svg);
+    Console.WriteLine(svg);
+		    return;
+		break;
+                case "/juliaswitch.js": svg = await webpage.Juliaswitch(svg, _env, "#734f96");
+                    await context.Response.WriteAsync(svg);
+                    return;
+                case "/button.js": svg = await button.SVG("{{contents}}", _env);
+                    await context.Response.WriteAsync(svg);
+                    return;
+		default:
+    Console.WriteLine("it's this default thing");
+    Console.WriteLine(context.Request.Path.Value);
+		break;
+	    };
+	
+        }
+            await _next(context);
+
+/*        if (context.Request.Path.Value.EndsWith("/foo.js"))
+        {
+            context.Response.ContentType = "application/javascript";
+	    string jsContent = await ReadJavaScriptFileAsync("osf-logo.js");
+	    string modifiedContent = await ModifyJavaScriptContent(jsContent);
+	    await context.Response.WriteAsync(modifiedContent);
+	    return;
+        }
+
+        await _next(context);*/
+    }
+
+    private async Task<string> ReadJavaScriptFileAsync(string fileName)
+    {
+	string filePath = Path.Combine(_env.WebRootPath, fileName);
+	if (!File.Exists(filePath))
+	{
+	    throw new FileNotFoundException($"The file {fileName} was not found in wwwroot.");
+	}
+	return await File.ReadAllTextAsync(filePath);
+    }
+
+    private string GenerateDynamicJavaScript()
+    {
+        // Generate your dynamic JavaScript content here
+        return "console.log('This is a dynamic JavaScript response!');";
+    }
+
+}
+
+// Extension method used to add the middleware to the HTTP request pipeline.
+public static class DynamicJavaScriptMiddlewareExtensions
+{
+    public static IApplicationBuilder UseDynamicJavaScript(this IApplicationBuilder builder)
+    {
+        return builder.UseMiddleware<DynamicJavaScriptMiddleware>();
+    }
```

Comment:
IMO lose this method. These read basically identical:
`builder.UseMiddleware<DynamicJavaScriptMiddleware>();`
`builder.UseDynamicJavascript();`